src/main/java/svc/data/citations/datasources/tyler/TylerTransformer.java

```diff
@@ -1,5 +1,154 @@
+
+		Citation output = new Citation();
+		output.citation_number = tylerCitation.citationNumber;
+		output.first_name = tylerCitation.firstName;
+		output.last_name = tylerCitation.lastName;
+		output.date_of_birth = LocalDate.parse(tylerCitation.dob, localDateFormatter);
+		output.drivers_license_number = tylerCitation.driversLicenseNumber;
+		output.citation_date = LocalDate.parse(tylerCitation.violationDate, localDateFormatter);
+
+		List<LocalDateTime> violationCourtDates = null;
+		violationCourtDates = tylerCitation.violations.stream()
+				.map((violation) -> violation.courtDate)
+				.distinct()
+				.map(this::parseViolationCourtDate)
+				.collect(Collectors.toList());
+		output.court_dateTime = violationCourtDates.size() > 0 ? violationCourtDates.get(0) : null;
+
+		output.violations = violationsFromTylerCitation(tylerCitation);
+		output.court_id = lookupCourtId(getTylerCourtIdentifier(tylerCitation));
+
+		// These could probably be added to the Tyler API
+		// citation time?
+		// Boolean mandatory_court_apperarnce
+		// Boolean can_pay_online
+		// public String defendant_address;  - not in XML
+		// public String defendant_city; - not in XML
+		// public String defendant_state; - not in XML
+		// Nice to add to our api DL State.
+
+		// There is no property for muni_id, but there probably should be
+
+		return output;
+	}
+
+	public List<Violation> violationsFromTylerCitation(TylerCitation tylerCitation) {
+		if (tylerCitation.violations != null) {
+			return tylerCitation.violations.stream()
+					.map(tylerViolation -> violationFromTylerViolation(tylerCitation, tylerViolation))
+					.collect(Collectors.toList());
+		}
+		return null;
+	}
+
+	public Violation violationFromTylerViolation(TylerCitation tylerCitation, TylerViolation tylerViolation) {
+		Violation output = new Violation();
+
+		output.citation_number = tylerCitation.citationNumber;
+		output.violation_number = tylerViolation.violationNumber;
+		output.violation_description = tylerViolation.violationDescription;
+		output.warrant_status = tylerViolation.warrantStatus;
+		output.warrant_number = tylerViolation.warrantNumber;
+		output.fine_amount = BigDecimal.valueOf(tylerViolation.fineAmount);
+		output.court_cost = null;
+
+		TylerViolationStatus violationStatus = TylerViolationStatus.fromTylerViolationStatusString(tylerViolation.status);
+		output.status = violationStatus != null ? violationStatus.getViolationStatus() : null;
+
+		// These might be available in the Tyler API, but they aren't piped
+		// through currently
+		// public LocalDate status_date;  - removed from this api
+		// public BigDecimal court_cost;
+
+		return output;
+	}
+	
+	private String getTylerCourtIdentifier(TylerCitation tylerCitation){
+		String tylerCourtIdentifier = null;
+		if (tylerCitation.violations != null && tylerCitation.violations.size() > 0){
+			for(TylerViolation tylerViolation: tylerCitation.violations){
+				tylerCourtIdentifier = tylerViolation.courtName;
```

Comment:
This logic seems flawed. There is no conditional here so it appears that the `tylerCourtIdentifier` will always be set to the first `tylerViolation`'s `courtName` in the array...